client/src/NewVideo.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState } from "react";
+
+export default function NewVivdeo({ setRefreshVideos }) {
+  const [formData, setFormData] = useState({
+    title: "",
+    url: "",
+  });
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    if (formData.title.trim() === "" || formData.url.trim() === "") {
+      alert("Title and URL cannot be empty");
+      return;
+    }
+
+    const formDataUrlChanged = {
+      title: formData.title,
+      url: `https://www.youtube-nocookie.com/embed/${
+        formData.url.match(/v=([a-zA-Z0-9_-]{11})/)[1]
+      }`,
```

Comment:
Can you talk through what this code is doing?

What assumptions are you making, and how confident are you in them? What could go wrong with it if your assumptions are violated?

Reply:
It's a regex that returns the first match, which is eleven characters after v= sign 
There was a thought that link could have params and queries, depending on user's action.
So that code is preventing this possibility from the very beginning.